Yoast/wordpress-seo
# Introduce an Upgrade History handler

## Summary

This PR can be summarized in the following changelog entry:

* Introduces a mechanism to save settings before executing an internal upgrade routine.

## Relevant technical choices:

* I had doubts about setting the key to `time()` instead of `current version`, but have decided to go for `time()` for the following arguments:
    - As we are an SEO plugin, time is a more prominent metric to determine if we need to act on changes from the past
    - In our testing plugin we allow to switch back to a previous version, this would be overwritten when only using a `current version`-index

## Test instructions

This PR can be tested by following these steps:

Using tools:
* Use the Yoast SEO Test Helper plugin to switch back to an older version, which will trigger the upgrade routine
* Open the database and see the `wpseo_upgrade_history` entry appear in the `wp_options` table, this should contain the data of the settings that have been stored in the other options. Also including the version of the previous Yoast SEO installation and the version of the current version.

Using older version:
* Create a new site (to make sure none of our options are in the database)
* Install Yoast SEO < 7.9
* Enter some settings, to make sure options are present in the database
* Install the new version
* Open the database and see the `wpseo_upgrade_history` entry appear in the `wp_options` table, this should contain the data of the settings that have been stored in the other options. Also including the version of the previous Yoast SEO installation and the version of the current version.

## Quality assurance


Fixes #10385
